src/indra_cogex/sources/clinicaltrials/__init__.py

```diff
@@ -75,25 +76,36 @@ def get_nodes(self):
                     self.has_trial_nct.append(row["NCTId"])
                     self.has_trial_cond_ns.append(cond_term.db)
                     self.has_trial_cond_id.append(cond_term.id)
-                    yield Node(
-                        db_ns=cond_term.db, db_id=cond_term.id, labels=["BioEntity"]
-                    )
                     found_disease_gilda = True
+                    if (cond_term.db, cond_term.id) not in yielded_nodes:
+                        yield Node(
+                            db_ns=cond_term.db,
+                            db_id=cond_term.id,
+                            labels=["BioEntity"],
+                            data=dict(name=cond_term.entry_name),
+                        )
+                        yielded_nodes.add((cond_term.db, cond_term.id))
             if not found_disease_gilda and not pd.isna(row["ConditionMeshId"]):
                 for mesh_id, mesh_term in zip(
                     row["ConditionMeshId"].split("|"),
                     row["ConditionMeshTerm"].split("|"),
                 ):
-                    correct_mesh_id = get_correct_mesh_id(mesh_id, mesh_term)
+                    correct_mesh_id, name = get_correct_mesh_id(mesh_id,
+                                                                mesh_term)
                     if not correct_mesh_id:
                         self.problematic_mesh_ids.append((mesh_id, mesh_term))
                         continue
                     self.has_trial_nct.append(row["NCTId"])
                     self.has_trial_cond_ns.append("MESH")
                     self.has_trial_cond_id.append(correct_mesh_id)
-                    yield Node(
-                        db_ns="MESH", db_id=correct_mesh_id, labels=["BioEntity"]
-                    )
+                    if ("MESH", correct_mesh_id) not in yielded_nodes:
```

Comment:
Standardization here might make sense

Reply:
Resolved in https://github.com/gyorilab/indra_cogex/pull/149/commits/e4eb11f697b20d98649fb44760c88a1648239efb